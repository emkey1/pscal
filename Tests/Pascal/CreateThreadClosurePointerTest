program CreateThreadClosurePointerTest;
uses Threading;

type
  PInteger = ^Integer;

procedure Run;
var
  base: integer;
  data: PInteger;
  tid: integer;
  status: integer;

  procedure Worker(param: pointer);
  var
    p: PInteger;
  begin
    p := PInteger(param);
    if p <> nil then
      p^ := p^ + base;
  end;

begin
  base := 5;
  New(data);
  data^ := 37;
  tid := CreateThread(@Worker, data);
  status := WaitForThread(tid);
  writeln('status=', status);
  writeln('heap=', data^);
  writeln('stats=', ThreadStatsCount);
  Dispose(data);
end;

begin
  Run;
end.
