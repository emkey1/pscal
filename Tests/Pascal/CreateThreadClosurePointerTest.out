status=0
heap=42
stats=0
